server/src/transactions/transactions.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CreateTransactionDto } from './dto/create-transaction.dto';
+import { Transaction } from './entities/transaction.entity';
+import { GetTransactionsFilterDto } from './dto/get-transactions-filter.dto';
+import { TransactionStatus } from './transaction-status.enum';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class TransactionsService {
+  constructor(
+    @InjectRepository(Transaction)
+    private transactionsRepository: Repository<Transaction>,
+  ) {}
+
+  async createTransaction(
+    createTransactionDto: CreateTransactionDto,
+  ): Promise<Transaction> {
+    return await this.transactionsRepository.save(createTransactionDto);
+  }
+
+  // look at this filter
+  getTransactions(filterDto: GetTransactionsFilterDto): Promise<Transaction[]> {
+    return this.transactionsRepository.find();
+  }
+
+  async getTransactionById(id: number): Promise<Transaction> {
+    // NOTE:  findOne is a built in function
+    const found = await this.transactionsRepository.findOne(id);
+    if (!found) {
+      throw new NotFoundException();
+    }
+    return found;
+  }
+
+  async updateTransaction(
+    id: number,
+    status: TransactionStatus,
+  ): Promise<Transaction> {
+    const transaction = await this.getTransactionById(id);
+    transaction.status = status;
+    await this.transactionsRepository.save(transaction);
+    return transaction;
+  }
+
+  async deleteTransaction(id: number): Promise<void> {
```

Comment:
i'm actually not sure what this is but let's figure it out and decide on a pattern right now
https://github.com/Nonprofit-Exchange-Hub/web-app/blob/57ab43bb106ee8b11105fcd2cd5a9a42ffd9271c/server/src/assets/assets.service.ts#L30
we do return something from this delete